pkg-builder.pl

```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use Config;
+use Cwd;
+use Data::Dumper;
+use File::Basename;
+use File::Copy;
+use File::Path qw/make_path/;
+use Getopt::Long;
+use IPC::Cmd qw/run can_run/;
+use Term::ANSIColor;
+
+my %DEFINES = ();
+
+sub parse_defines()
+{
+   Die("wrong commandline options")
+     if ( !GetOptions( "defines=s" => \%DEFINES ) );
+
+   Die("jetty.distro not specified (--define jetty.distro=<xxxx>)")
+     if ( !exists $DEFINES{'jetty.distro'} );
+}
+
+sub cpy_file($$)
+{
+   my $src_file = shift;
+   my $des_file = shift;
+
+   my $des_dir = dirname($des_file);
+
+   make_path($des_dir)
+     if ( !-d $des_dir );
+
+   Die("copy '$src_file' -> '$des_file' failed!")
+     if ( !copy( $src_file, $des_file ) );
+}
+
+sub git_timestamp_from_dirs($)
+{
+   my $dirs = shift || [];
+
+   print Dumper($dirs);
+
+   my $ts;
+   if ( $dirs && @$dirs )
+   {
+      foreach my $dir (@$dirs)
+      {
+         chomp( my $ts_new = `git log --pretty=format:%at -1 '$dir'` );
+         Die("failed to get git timestamp from $dir")
+           if ( !defined $ts_new );
+         $ts = $ts_new
+           if ( !defined $ts || $ts_new > $ts );
+      }
+   }
+
+   return $ts;
+}
+
+
+my %PKG_GRAPH = (
+   "zimbra-timezone-data" => {
+      summary    => "Zimbra Timezone Data",
+      version    => "1.0.0",
+      revision   => 1,
+      hard_deps  => [],
+      soft_deps  => [],
+      other_deps => ["zimbra-core-components"],
+      replaces   => [ "zimbra-store", "zimbra-core" ],
+      file_list  => ['/opt/zimbra/*'],
+      stage_fun  => sub { &stage_zimbra_timezone_data(@_); },
+   },
+);
+
+
+sub stage_zimbra_timezone_data()
+{
+   my $stage_base_dir = shift;
+
+   foreach my $webapp_name ( "zimbra", "zimbraAdmin" )
+   {
+      cpy_file( "build/js/AjxTimezoneData.js",           "$stage_base_dir/opt/zimbra/$DEFINES{'jetty.distro'}/webapps/$webapp_name/js/ajax/util/AjxTimezoneData.js" );
```

Comment:
I don't know about the 'split-ui' setup. To selectively install files, we would need to split the package to be installed into multiple (one per deployment type). This is going to create unnecessary complexity and with additional supporting changes in the installer.  The current packaging is not much different than the situation that we already are with the files being bundled with zimbra-store and zimbra-core packages, which are required on most deployments. However the a split in packages (one per deployment type) is not precluded in the future, we can streamline it as required in the future.